examples/python/lattice/hexagonal/build-xml.py

```diff
@@ -0,0 +1,192 @@
+moderator.add_nuclide(o16, 1.)
+moderator.add_s_alpha_beta('HH2O', '71t')
+
+# Instantiate a MaterialsFile, register all Materials, and export to XML
+materials_file = openmc.MaterialsFile()
+materials_file.set_default_xs('71c')
+materials_file.add_materials([moderator, fuel])
+materials_file.export_to_xml()
+
+
+###############################################################################
+#                 Exporting to OpenMC geometry.xml File
+###############################################################################
+
+# Instantiate Surfaces
+left = openmc.XPlane(surface_id=1, x0=-2, name='left')
+right = openmc.XPlane(surface_id=2, x0=2, name='right')
+bottom = openmc.YPlane(surface_id=3, y0=-2, name='bottom')
+top = openmc.YPlane(surface_id=4, y0=2, name='top')
+fuel1 = openmc.ZCylinder(surface_id=5, x0=0, y0=0, R=0.4)
+fuel2 = openmc.ZCylinder(surface_id=6, x0=0, y0=0, R=0.3)
+fuel3 = openmc.ZCylinder(surface_id=7, x0=0, y0=0, R=0.2)
+
+left.set_boundary_type('vacuum')
+right.set_boundary_type('vacuum')
+top.set_boundary_type('vacuum')
+bottom.set_boundary_type('vacuum')
+
+# Instantiate Cells
+cell1 = openmc.Cell(cell_id=1, name='Cell 1')
+cell2 = openmc.Cell(cell_id=101, name='cell 2')
+cell3 = openmc.Cell(cell_id=102, name='cell 3')
+cell4 = openmc.Cell(cell_id=201, name='cell 4')
+cell5 = openmc.Cell(cell_id=202, name='cell 5')
+cell6 = openmc.Cell(cell_id=301, name='cell 6')
+cell7 = openmc.Cell(cell_id=302, name='cell 7')
+
+# Register Surfaces with Cells
+cell1.add_surface(left, halfspace=+1)
+cell1.add_surface(right, halfspace=-1)
+cell1.add_surface(bottom, halfspace=+1)
+cell1.add_surface(top, halfspace=-1)
+cell2.add_surface(fuel1, halfspace=-1)
+cell3.add_surface(fuel1, halfspace=+1)
+cell4.add_surface(fuel2, halfspace=-1)
+cell5.add_surface(fuel2, halfspace=+1)
+cell6.add_surface(fuel3, halfspace=-1)
+cell7.add_surface(fuel3, halfspace=+1)
+
+# Register Materials with Cells
+cell2.set_fill(fuel)
+cell3.set_fill(moderator)
+cell4.set_fill(fuel)
+cell5.set_fill(moderator)
+cell6.set_fill(fuel)
+cell7.set_fill(moderator)
+
+# Instantiate Universe
+univ1 = openmc.Universe(universe_id=1)
+univ2 = openmc.Universe(universe_id=2)
+univ3 = openmc.Universe(universe_id=3)
+root = openmc.Universe(universe_id=0, name='root universe')
+
+# Register Cells with Universe
+univ1.add_cells([cell2, cell3])
+univ2.add_cells([cell4, cell5])
+univ3.add_cells([cell6, cell7])
+root.add_cell(cell1)
+
+# Instantiate a Lattice
+lattice = openmc.HexLattice(lattice_id=5)
+lattice.set_num_rings(3)
+#lattice.set_center([0., 0.])
+#lattice.set_pitch([1.])
+#lattice.set_universes([
+#     [univ2, univ2, univ2, univ2, univ2, univ2, univ2, univ2, univ2, univ2, univ2, univ2],
+#     [univ2, univ2, univ2, univ2, univ2, univ2],
+#     [univ1]])
+lattice.set_center([0., 0., 0.])
+lattice.set_pitch([1., 1.])
+lattice.set_num_axial(2)
+lattice.set_universes([
+     [
+      [univ2, univ2, univ2, univ2, univ2, univ2, univ2, univ2, univ2, univ2, univ2, univ2],
```

Comment:
You might use Python's list comprehension to simplify this, which I think is one reason to use "ring" notation anyway. Perhaps something like:

```
lattice.set_center([0., 0., 0.])
lattice.set_pitch([1., 1.])
lattice.set_num_axial(2)
lattice.set_universes([
     [
      [univ2]*12,
      [univ2]*6,
      [univ1]
     ],
     [
      [univ2]*12,
      [univ2]*6,
      [univ1]
     ]
     ])
```